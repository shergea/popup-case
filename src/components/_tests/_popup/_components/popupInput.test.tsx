import { Formik } from "formik";
import renderer from "react-test-renderer";
import PopupInput from "../../../popup/components/popupInput";

it("renders correctly", () => {
  const fieldName = "firstName";
  const labelName = "First Name";
  const tree = renderer
    .create(
      <Formik
        validate={(values: any) => {
          const errors = {} as any;

          if (!values?.firstName) {
            errors.firstName = "Required.";
          }

          return errors;
        }}
        initialValues={{}}
        onSubmit={() => {}}
      >
        <PopupInput
          type="text"
          id={fieldName}
          name={fieldName}
          placeholder={labelName}
        />
      </Formik>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
