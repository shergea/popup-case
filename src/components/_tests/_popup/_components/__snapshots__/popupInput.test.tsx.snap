// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="mt-4"
>
  <input
    className="border border-gray-300 bg-gray-100 text-base font-medium block w-full p-2.5 placeholder:text-black placeholder:text-base placeholder:font-medium outline-0"
    id="firstName"
    name="firstName"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="First Name"
    type="text"
  />
</div>
`;
